src/components/accession2/DetailPanel.tsx

```diff
@@ -0,0 +1,142 @@
+import { TabContext, TabList, TabPanel } from '@mui/lab';
+import { Box, Grid, Tab, Typography, useTheme } from '@mui/material';
+import moment from 'moment';
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { Accession2, getAccession2 } from 'src/api/accessions2/accession';
+import strings from 'src/strings';
+import useDeviceInfo from 'src/utils/useDeviceInfo';
+import TfMain from '../common/TfMain';
+
+type DetailPanelProps = {
+  accession?: Accession2;
+};
+export default function DetailPanel(props: DetailPanelProps): JSX.Element {
+  const { accession } = props;
+  const theme = useTheme();
+  const { isMobile } = useDeviceInfo();
+
+  const categoryStyle = {
+    color: '#708284',
+  };
+
+  const gridRowStyle = {
+    display: isMobile ? 'block' : 'flex',
+    padding: theme.spacing(2, 0),
+  };
+
+  const gridLeftSide = isMobile ? 12 : 4;
+
+  const gridRightSide = isMobile ? 12 : 8;
+  const today = moment();
+  const seedCollectionDate = accession?.collectedDate ? moment(accession?.collectedDate, ' YYYY-MM-DD') : undefined;
+
+  const age = seedCollectionDate ? today.diff(seedCollectionDate, 'months') : undefined;
+
+  console.log(age);
+
+  const displayAge = () => {
+    if (age! < 1) {
```

Comment:
looks like age can be undefined according to code above, we should maybe handle the undefined case?

Reply:
It is managed before calling this method

Reply:
Hmm, doesn't make the code very readable then.
Can we at least have a check for age and if it is undefined output some error like "this should not happen" and return empty string?
I think we should make it obvious with some code, otherwise only the person that wrote this code would know.